tests/lib/rules/apple-touch-icons/tests.ts

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'fs';
+
+import { generateHTMLPage } from '../../../helpers/misc';
+import { getRuleName } from '../../../../src/lib/utils/rule-helpers';
+import { IRuleTest } from '../../../helpers/rule-test-type'; // eslint-disable-line no-unused-vars
+import * as ruleRunner from '../../../helpers/rule-runner';
+
+const ruleName = getRuleName(__dirname);
+
+const defaultImage = fs.readFileSync(`${__dirname}/fixtures/apple-touch-icon.png`); // eslint-disable-line no-sync
+const imageWithIncorrectDimensions = fs.readFileSync(`${__dirname}/fixtures/incorrect-dimensions.png`); // eslint-disable-line no-sync
+const imageThatIsNotSquare = fs.readFileSync(`${__dirname}/fixtures/not-square.png`); // eslint-disable-line no-sync
+const imageWithIncorrectFileFormat = fs.readFileSync(`${__dirname}/fixtures/incorrect-file-format.png`); // eslint-disable-line no-sync
+// const imageWithTransparentBackground = fs.readFileSync(`${__dirname}/fixtures/transparent-background.png`); // eslint-disable-line no-sync
```

Comment:
Do you plan on adding the transparency part in this PR?

Reply:
> Do you plan on adding the transparency part in this PR?

No. That part is a bit tricky, so I'll try to do it once we have more time.

Reply:
Then we should add an issue for that. What is the problem when using transparency?

Reply:
> What is the problem when using transparency?

1) I want to void introducing a big image manipulation library just to check that.
2) I also want to figure out what should and should not pass (e.g.: you have just a few transparent pixels, it should be fine).

Reply:
> Then we should add an issue for that

[Done.](https://github.com/sonarwhal/sonar/issues/496)